src/adapter/src/coord.rs

```diff
@@ -473,6 +474,15 @@ impl PlanValidity {
                 return Err(AdapterError::ChangedPlan);
             }
         }
+        if catalog
+            .try_get_role(&self.role_metadata.current_role)
+            .is_none()
+            || catalog
+                .try_get_role(&self.role_metadata.session_role)
+                .is_none()
+        {
+            return Err(AdapterError::ChangedPlan);
```

Comment:
should this be AdapterError::Unauthorized ?

Reply:
Yeah, that probably works better, just updated.